### PR TITLE
feat(dpp): basic validtion for state transition

### PR DESCRIPTION
**Pull request by qrayven**
Basic validation of State Transition

## What was done?

basic validation of state transition with included dependecies

## How Has This Been Tested?

unit tests

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

### CODE REVIEW

File: dpp/src/state_transition/state_transition_factory.rs [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+        StateTransitionType::IdentityTopUp => {
+            let transition = IdentityTopUpTransition::new(raw_state_transition)?;
+            Ok(StateTransition::IdentityTopUp(transition))
+        }
+        StateTransitionType::IdentityCreditWithdrawal => {
+            let transition =
+                IdentityCreditWithdrawalTransition::from_raw_object(raw_state_transition)?;
+            Ok(StateTransition::IdentityCreditWithdrawal(transition))
+        }
+        StateTransitionType::DocumentsBatch => {
```
Comment (antouhou) [inline dpp/src/state_transition/state_transition_factory.rs:53]: Isn't that the same thing as in the `if` block in the beginning of the function?

File: dpp/src/state_transition/validation/validator_state_transition_identity_signature.rs [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+    let maybe_public_key = identity.get_public_key_by_id(signature_public_key_id);
+
+    let public_key = match maybe_public_key {
+        None => {
+            validation_result.add_error(SignatureError::MissingPublicKeyError {
+                public_key_id: signature_public_key_id,
+            });
+            return Ok(validation_result);
+        }
+        Some(id) => id,
```
Comment (antouhou) [inline dpp/src/state_transition/validation/validator_state_transition_identity_signature.rs]: Maybe rename it to `pk` instead of `id`? :)

**Review by antouhou — COMMENTED** [review]
I've left two small comments, please take a look. I like the way the example state transition mock is done, good job!

File: dpp/src/state_transition/state_transition_factory.rs (hunk 1 above)
Comment (qrayven) [reply dpp/src/state_transition/state_transition_factory.rs:53]: yes it is! I forgot to remove the first one after refactor, thanks!

File: dpp/src/state_transition/validation/validator_state_transition_identity_signature.rs (hunk 2 above)
Comment (qrayven) [reply dpp/src/state_transition/validation/validator_state_transition_identity_signature.rs]: good point, fixed

**Review by antouhou — APPROVED** [review]
👍

**Pull request merged**